src/posit/connect/content.py

```diff
@@ -119,76 +119,108 @@ def restart(self) -> None:
+        min_processes : int, optional
+            Min concurrent processes required. Default is None. Falls back to server setting 'Scheduler.MinProcesses'.
+        max_conns_per_process : int, optional
+            Max client connections per process. Default is None. Falls back to server setting 'Scheduler.MaxConnsPerProcess'.
+        load_factor : float, optional
+            Aggressiveness in spawning new processes (0.0 - 1.0). Default is None. Falls back to server setting 'Scheduler.LoadFactor'.
+        cpu_request : float, optional
+            Min CPU units required (1 unit = 1 core). Default is None. Falls back to server setting 'Scheduler.CPURequest'.
+        cpu_limit : float, optional
+            Max CPU units allowed. Default is None. Falls back to server setting 'Scheduler.CPULimit'.
+        memory_request : int, optional
+            Min memory (bytes) required. Default is None. Falls back to server setting 'Scheduler.MemoryRequest'.
+        memory_limit : int, optional
+            Max memory (bytes) allowed. Default is None. Falls back to server setting 'Scheduler.MemoryLimit'.
+        amd_gpu_limit : int, optional
+            Number of AMD GPUs allocated. Default is None. Falls back to server setting 'Scheduler.AMDGPULimit'.
+        nvidia_gpu_limit : int, optional
+            Number of NVIDIA GPUs allocated. Default is None. Falls back to server setting 'Scheduler.NvidiaGPULimit'.
+        run_as : str, optional
+            UNIX user to execute the content. Default is None. Falls back to server setting 'Applications.RunAs'.
+        run_as_current_user : bool, optional
+            Run process as the visiting user (for app content). Default is False.
+        default_image_name : str, optional
+            Default image for execution if not defined in the bundle. Default is None.
+        default_r_environment_management : bool, optional
+            Manage R environment for the content. Default is None.
+        default_py_environment_management : bool, optional
+            Manage Python environment for the content. Default is None.
+        service_account_name : str, optional
+            Kubernetes service account name for running content. Default is None.
+
+        Returns
+        -------
+        None
         """
         ...
 
     @overload
-    def update(self, *args, **kwargs) -> None:
-        """Update the content item."""
+    def update(self, **attributes: Any) -> None:
+        """Update the content."""
         ...
 
-    def update(self, *args, **kwargs) -> None:
-        """Update the content item."""
-        body = dict(*args, **kwargs)
-        url = self.params.url + f"v1/content/{self.guid}"
-        response = self.params.session.patch(url, json=body)
+    def update(self, **attributes: Any) -> None:
+        """Update the content."""
+        url = self.params.url + f"v1/content/{self['guid']}"
```

Comment:
not sure how to weigh pros / cons here, just an observation - when I'm interacting with an SDK resource, my LSP would give me autocomplete on the deprecated property fields, which I don't get using the new approach. Its harder to explore what fields are available on the resource, or in some cases know where to look at all for that information if it isn't in a docstring.  

Its also easier for the LSP to catch typos on properties - if I were to typo a dict key, I would get a more confusing error.

Reply:
Good callout. That is a big downside to this approach and the main reason we tried using attributes originally. I have experimented with defining a TypedDict and then holding data in an internal attribute instead of on self. I abandoned it due to the complexity of maintaining the facade between the internal state and self. I'm open to revisiting it, though.

The other answer is... you always have that problem when using dictionaries in Python. So this isn't anything new  🙃

Reply:
yeah, it might just be a matter of making sure we're good with our documentation of resources so we can easily check what's in them.